src/Compilers/Core/Portable/NativePdbWriter/PdbWriter.cs

```diff
@@ -68,9 +72,9 @@ internal PdbLogger(bool logging)
             }
         }
 
-        private void MaybeFlush()
+        private void EnsureSpace(int space)
         {
-            if (_logData.Count >= bufferFlushLimit)
+            if (_logData.Count + space >= bufferFlushLimit)
```

Comment:
It should not be resized after this change (in theory). Perhaps I should try using BlobWriter